tools/hermes-parser/js/flow-api-translator/src/flowToFlowDef.js

```diff
@@ -986,23 +986,43 @@ function convertSuperClass(
     return EMPTY_TRANSLATION_RESULT;
   }
 
-  if (superClass.type !== 'Identifier') {
-    throw translationError(
-      superClass,
-      `SuperClass: Non identifier super type of "${superClass.type}" not supported`,
-      context,
-    );
+  switch (superClass.type) {
+    case 'Identifier':
+    case 'MemberExpression': {
+      const [resultTypeParams, typeParamsDeps] =
+        convertTypeParameterInstantiationOrNull(superTypeParameters, context);
+      const superDeps = analyzeTypeDependencies(superClass, context);
+      return [
+        t.InterfaceExtends({
+          id: asDetachedNode(superClass),
+          typeParameters: resultTypeParams,
+        }),
+        [...typeParamsDeps, ...superDeps],
+      ];
+    }
+    case 'TypeCastExpression': {
```

Comment:
Its not safe to pass through the `TypeAnnotationType` because only `QualifiedTypeIdentifier` nodes are supported, nodes like `GenericTypeAnnotation` need to be converted to Identifiers.